bench/wtperf/config.c

```diff
@@ -438,14 +438,13 @@ config_opt(WTPERF *wtperf, WT_CONFIG_ITEM *k, WT_CONFIG_ITEM *v)
 			return (EINVAL);
 		}
 		strp = (char **)valueloc;
-		newlen = v->len + 1;
 		if (*strp == NULL)
 			begin = newstr = dstrdup(v->str);
 		else {
-			newlen += strlen(*strp) + 1;
-			newstr = dcalloc(newlen, sizeof(char));
-			snprintf(newstr, newlen,
-			    "%s,%*s", *strp, (int)v->len, v->str);
+			newlen = strlen(*strp) + v->len + strlen(",") + 1;
+			newstr = dmalloc(newlen);
+			testutil_check(__wt_snprintf(newstr, newlen,
+			    "%s,%.*s", *strp, (int)v->len, v->str));
```

Comment:
Note the change from `%*s` to `%.*s`, this is a bug fix, additionally, the length of the allocated memory is too short and fails in testing.